### PR TITLE
support for proxy url option for use with react native web

### PR DESCRIPTION
**Pull request by megantaylor**
to resolve https://github.com/ospfranco/link-preview-js/issues/69

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -260,7 +260,7 @@ export async function getLinkPreview(
     throw new Error(`link-preview-js did not receive a valid url or text`);
   }
 
-  const detectedUrl = text
+  const detectedUrl = (options?.proxyUrl ? options.proxyUrl.concat(text) : text)
```
Comment (ospfranco) [inline index.ts:263]: can you explain how is this supposed to work? the entire text gets concatenated at the end of the proxyUrl, but then it gets spliced? 

Can you also add a test for this?

**Review by ospfranco — CHANGES_REQUESTED** [review]

File: index.ts (hunk 1 above)
Comment (megantaylor) [reply index.ts:263]: oh, i think i misunderstood how detectedUrl works. the idea was to concat the proxyUrl + url for the fetch request. i just need to move the logic, i think.

Comment (megantaylor) [reply index.ts]: moved the logic, need to think about a good test

**Comment by megantaylor** [comment]
hmm, any idea why the travis ci build failed?

**Comment by ospfranco** [comment]
TravisCI is not properly configured, I just needed the build for another project, don't worry about it

File: index.ts [hunk 2]
```diff
@@ -260,21 +260,20 @@ export async function getLinkPreview(
...
     throw new Error(`link-preview-js did not receive a valid a url or text`);
   }
 
   const fetchOptions = { headers: options?.headers ?? {} };
 
+  const fetchUrl = options?.proxyUrl ? options.proxyUrl.concat(detectedUrl) : detectedUrl;
+
   try {
-    const response = await fetch(detectedUrl, fetchOptions);
+    const response = await fetch(fetchUrl, fetchOptions);
```
Comment (ospfranco) [inline index.ts:274]: looks better! now the only thing missing is the test 👍

**Review by ospfranco — APPROVED** [review]

File: index.ts (hunk 2 above)
Comment (megantaylor) [reply index.ts:274]: added a test

**Comment by ospfranco** [comment]
Cheers @megantaylor, I'll release a new version soon

**Comment by ospfranco** [comment]
released with version `2.0.5`

**Pull request merged**